src/renderable/sprite.js

```diff
@@ -262,12 +262,19 @@
             var w = this.width, h = this.height;
             var angle = this.angle + this._sourceAngle;
 
+            // save context
+            renderer.save();
+            
+            // calculate pixel pos of the anchor point
+            var ax = w * this.anchorPoint.x, ay = h * this.anchorPoint.y;
+            renderer.translate(-ax, -ay);
+            
+            // adjust geometry box position based on anchor point offset
+            this._leftOffset = xpos - ax;
+            this._topOffset = ypos - ay;
+            
             if ((this.scaleFlag) || (angle !== 0)) {
-                // save context
-                renderer.save();
 
-                // calculate pixel pos of the anchor point
-                var ax = w * this.anchorPoint.x, ay = h * this.anchorPoint.y;
                 // translate to the defined anchor point
                 renderer.translate(xpos + ax, ypos + ay);
```

Comment:
Yeah, crazy enough it's working. The sprite is rotating around its `pivotPoint`, which is placed at the Entity's `pivotPoint`.

I've found an undesirable behavior, though, that I believe is a pre-existing issue. Normally in game engines scale is applied before rotation so you are rotating a pre-scaled object. However, it looks like it's being done the other way around, and because of this the sprite deforms during rotation when it also has scale applied.